### PR TITLE
Sanitize issue level to match real world

### PR DESCRIPTION
**Pull request by La0**
Prep work for #364.

---

Use an explicit enum for issue levels, bringing an extra level (on top of error & warning) as **critical**.
The idea is to mark issues as critical when they come from a tier 1 analyzer (in a followup), and always report them.

By bringing that enum, i also noticed some discrepancies in our level usage (we don't _really_ use it actually):

- the clang-tidy parser used a level variable, that was never set (as the level is not set anymore in the current payload, and notes are parsed in mach) : so every clang-tidy is at Warning level.
- the infer payload only has `ERROR` or `WARNING`, not random strings.

### CODE REVIEW

File: backend/code_review_backend/issues/models.py [hunk 1]
```diff
@@ -11,7 +11,12 @@
 
 LEVEL_WARNING = "warning"
 LEVEL_ERROR = "error"
-ISSUE_LEVELS = ((LEVEL_WARNING, "Warning"), (LEVEL_ERROR, "Error"))
+LEVEL_CRITICAL = "critical"
+ISSUE_LEVELS = (
+    (LEVEL_WARNING, "Warning"),
+    (LEVEL_ERROR, "Error"),
+    (LEVEL_CRITICAL, "Critical"),
+)
```
Comment (marco-c) [inline backend/code_review_backend/issues/models.py]: What if, for simplicity, we keep only two levels: `error` and `warning`? `error` for CI breaking issues, `warning` for all other issues which developers can feel free to ignore.

Comment (marco-c) [reply backend/code_review_backend/issues/models.py]: And we'd temporarily force mozlint issues to be errors and all others to be warnings, until we fix it in mozilla-central

Comment (La0) [reply backend/code_review_backend/issues/models.py]: They already are errors ! that's why i added that extra level

Comment (marco-c) [reply backend/code_review_backend/issues/models.py]: Yep, but the others can be errors too at the moment, instead we could make them warnings and consider "error" as breaking CI and "warning" as not breaking CI. This way we reduce the changes we need to make (both here and in mozilla-central), without losing any feature (as we are currently not using the level basically).

Comment (La0) [reply backend/code_review_backend/issues/models.py]: Ok, works for me

Comment (La0) [reply backend/code_review_backend/issues/models.py]: Done, i simply removed the backend patch & the critical level (in 5fdfac8)

File: bot/code_review_bot/__init__.py [hunk 2]
```diff
@@ -57,7 +65,7 @@ def __init__(
         check: str,
         column: int = None,
         message: str = None,
-        level: str = "error",
+        level: Level = Level.Error,
```
Comment (marco-c) [inline bot/code_review_bot/__init__.py]: Maybe warning by default, so when we start publishing all error issues we don't risk publishing some we didn't have to publish

File: bot/code_review_bot/tasks/coverity.py [hunk 3]
```diff
@@ -54,7 +55,7 @@ def __init__(self, analyzer, revision, issue, file_path):
             line=issue["line"],
             nb_lines=1,
             check=issue["flag"],
-            level="error",
+            level=Level.Error,
```
Comment (marco-c) [inline bot/code_review_bot/tasks/coverity.py]: Coverity should be warnings, at least for now

File: bot/code_review_bot/tasks/infer.py [hunk 4]
```diff
@@ -48,13 +49,10 @@ def __init__(self, analyzer, entry, revision):
             nb_lines=1,
             check=entry["bug_type"],
             column=entry["column"],
-            level=kind,
+            level=Level(kind.lower()),
```
Comment (marco-c) [inline bot/code_review_bot/tasks/infer.py]: Infer too for now should be always warnings

File: bot/code_review_bot/__init__.py (hunk 2 above)
Comment (La0) [reply bot/code_review_bot/__init__.py]: Done.

File: bot/code_review_bot/tasks/coverity.py (hunk 3 above)
Comment (La0) [reply bot/code_review_bot/tasks/coverity.py]: Done.

File: bot/code_review_bot/tasks/infer.py (hunk 4 above)
Comment (La0) [reply bot/code_review_bot/tasks/infer.py]: Done.

File: bot/tests/test_infer.py [hunk 5]
```diff
@@ -106,5 +106,7 @@ def test_infer_artifact(version, nb, mock_revision, mock_hgmo):
 
     issues_data = [issue.as_dict() for issue in issues]
 
+    with open(os.path.join(FIXTURES_DIR, f"infer_issues_{version}.json"), "w") as f:
```
Comment (marco-c) [inline bot/tests/test_infer.py]: I guess this is a left over ;)

Comment (La0) [reply bot/tests/test_infer.py]: woops

**Pull request merged**